pyautomagic/preprocessing/prep/noisychannels.py

```diff
@@ -0,0 +1,383 @@
+                           copy=True,phase="zero",fir_window="hamming",fir_design="firwin",pad="reflect_limited",
+                           verbose=None)
+    EEGData = signal.detrend(EEGData)
+    # removing line noise
+    mne.filter.notch_filter(EEGData,sample_rate,linenoise,filter_length="auto",
+                            notch_widths=None,trans_bandwidth=1,method="fir",iir_params=None,mt_bandwidth=None,
+                            p_value=0.05,picks=None,n_jobs=1,copy=True,phase="zero",fir_window="hamming",
+                            fir_design="firwin",pad="reflect_limited",verbose=None)
+    # finding channels with NaNs or constant values for long periods of time
+    original_dimensions = np.shape(EEGData)
+    original_channels = np.arange(original_dimensions[0])
+    channels_interpolate = original_channels
+    nan_channel_mask = [False] * original_dimensions[0]
+    no_signal_channel_mask = [False] * original_dimensions[0]
+
+    for i in range(0, original_dimensions[0]):
+        nan_channel_mask[i] = np.sum(np.isnan(EEGData[i, :])) > 0
+    for i in range(0, original_dimensions[0]):
+        no_signal_channel_mask[i] = robust.mad(EEGData[i, :]) < 10 ** (-10) or np.std(
+            EEGData[i, :]) < 10 ** (-10)
+    nan_channels = channels_interpolate[nan_channel_mask]
+    no_data_channels = channels_interpolate[no_signal_channel_mask]
+    for i in range(0, original_dimensions[0]):
+        if nan_channel_mask[i] == True or no_signal_channel_mask[i] == True:
+            EEGData = np.delete(EEGData, i, axis=0)
+    nans_no_data_channels = np.union1d(nan_channels, no_data_channels)
+    channels_interpolate = np.setdiff1d(
+    channels_interpolate, nans_no_data_channels)  # channels to be used for interpolation
+    nans_no_data_ChannelName = list()
+    ch_names = raw.info["ch_names"]
+    for i in range(0, len(nans_no_data_channels)):
+        nans_no_data_ChannelName.append(ch_names[nans_no_data_channels[i]])
+    raw.drop_channels(nans_no_data_ChannelName)
+    evaluation_channels = channels_interpolate
+    new_dimension = np.shape(EEGData)
+
+    # find channels that have abnormally high or low amplitude
+    robust_channel_deviation = np.zeros(original_dimensions[0])
+    deviation_channel_mask = [False] * (new_dimension[0])
+    channel_deviation = np.zeros(new_dimension[0])
+    for i in range(0, new_dimension[0]):
+        channel_deviation[i] = 0.7413 * iqr(EEGData[i, :])
+    channel_deviationSD = 0.7413 * iqr(channel_deviation)
+    channel_deviationMedian = np.nanmedian(channel_deviation)
+    robust_channel_deviation[evaluation_channels] = np.divide(
+        np.subtract(channel_deviation, channel_deviationMedian), channel_deviationSD
+    )
+    for i in range(0, new_dimension[0]):
+        deviation_channel_mask[i] = abs(robust_channel_deviation[i]) > 5 or np.isnan(
+            robust_channel_deviation[i]
+        )

```

Comment:
Dimensions seems not matching, robust_channel_deviation has original_dimension but i only goes through new_dimension